src/app-config/posthog.ts

```diff
@@ -1,25 +1,56 @@
+import fs from 'fs-extra';
 import { PostHog, PostHogOptions } from 'posthog-node';
+import { v4 as uuidv4 } from 'uuid';
 
-type PostHogCliOptions = PostHogOptions & {
-  analyticsId: string;
-};
+type PostHogCliOptions = Omit<PostHogOptions, 'persistence'> & ({
+  persistence: 'memory'
+} | {
+  persistence: 'file';
+  propertiesFile: string;
+} | Record<string, never>);
 
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-ignore
 export class PostHogCli extends PostHog {
-  constructor(apiKey: string, private options: PostHogCliOptions) {
-    super(apiKey, options);
+  private _properties: Record<string, any | undefined>;
+
+  constructor(apiKey: string, private _options: PostHogCliOptions) {
+    super(apiKey, {
+      ..._options,
+      persistence: 'memory',
+    });
+
+    this._properties = {};
+    if (this._options.persistence === 'file' && fs.existsSync(this._options.propertiesFile)) {
+      this._properties = fs.readJSONSync(this._options.propertiesFile);
+    } else {
+      this.setPersistedProperty('anonymous_id', uuidv4());
+    }
   }
 
-  capture(message: { event: string; properties?: Record<string | number, any>; }): void {
-    // We override reInit so the distinctId isn't used
-    return super.capture({ distinctId: '<unused>', ...message });
+  /**
+   * @override
+   */
+  getPersistedProperty(key: string): any | undefined {
```

Comment:
Hmm I'm super confused. We are extending the PostHog node class.
```
export class PostHog implements PostHogNodeV1 {
  private _sharedClient: PostHogClient
}
```

`getPersistedProperty` and `setPersistedProperty` are functions on the `_sharedClient`. So these would be net new methods not overrides unless I'm missing something.

Reply:
Ah I see the new package changed it